File: client/scalaz/src/main/scala/org/openapitools/client/api/SubjectList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import SubjectList._

case class SubjectList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Subject]])

object SubjectList {
  import DateTimeCodecs._

  implicit val SubjectListCodecJson: CodecJson[SubjectList] = CodecJson.derive[SubjectList]
  implicit val SubjectListDecoder: EntityDecoder[SubjectList] = jsonOf[SubjectList]
  implicit val SubjectListEncoder: EntityEncoder[SubjectList] = jsonEncoderOf[SubjectList]
}

File: client/javascript-closure-angular/API/Client/SubjectList.js
goog.provide('API.Client.SubjectList');

/**
 * @record
 */
API.Client.SubjectList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.SubjectList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.SubjectList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Subject>}
 * @export
 */
API.Client.SubjectList.prototype.data;

